Home page design

SupaHotBall opened the pull request:
## Description

- Implemented dashboard-style Home page layout with two key sections:
  - IT Jobs by Country (city-based job distribution)
  - Skills (bar chart of job skill usage)
- Added mock data for both sections as placeholders before backend integration
- Ensured Home page loads immediately with visual components styled for clarity

## Changes Made

- Created `Home.jsx` as the new landing page
- Added `ITJobsByCountryCard.jsx` and `SkillsChart.jsx` components
- Installed and configured `react-router-dom` and `recharts`
- Updated route structure in `App.jsx` to support the new Home
- Centralized all styles in `App.css` with modern layout using CSS grid
- Fixed layout bugs and ensured proper component alignment and appearance


![{BD76D47E-7409-4779-860C-E54F6EA80C4D}](https://github.com/user-attachments/assets/417e7abc-3247-4335-91d2-4854c4c9c995)

![{90464F97-6804-47CD-B840-080E849D4893}](https://github.com/user-attachments/assets/8c88ad22-0f39-4830-ba51-3bb4cc394227)

SupaHotBall committed:
Feat: Create a file in components called ITJobsByCountryCard.jsx

SupaHotBall committed:
Feat: Added the logic for the ITJobsByCountry

SupaHotBall committed:
feat: created a file called SkillsChart.jsx

SupaHotBall committed:
Feat: Added a SkillsChart logic for the home page

SupaHotBall committed:
Feat: Edited the return in Home.jsx

SupaHotBall committed:
Feat: change some css for the home page

SupaHotBall committed:
Feat: Add stacked skills categories to Home dashboard layout

AhmadShoaib1 reviewed (approved):
## 🧾 Pull Request Review – Enhancing Home Page with Skills & Country Cards

### ✅ What's Good

- **✅ Modular Components**: `ITJobsByCountryCard.jsx` and `SkillsChart.jsx` are nicely separated and scoped. This modularity makes the codebase cleaner and easier to maintain.
- **✅ Visual Enhancement**: Integrating data visualizations via Recharts significantly improves the user interface and experience.
- **✅ Tailwind Consistency**: Great to see consistent use of Tailwind CSS throughout – it keeps the UI unified.
- **✅ Clean Home Page Integration**: The new components are well-integrated into `HomePage.jsx`, with no clutter.

---

### Suggestions for Improvement

####  Error Handling & Loading States

Neither component currently handles loading, error, or empty data states.

**Suggested Improvement**:
```jsx
if (isLoading) return <div>Loading...</div>;
if (error) return <div>Error loading data.</div>;
if (!data?.length) return <div>No data available.</div>;

AhmadShoaib1 reviewed (changes requested):
## 🧾 Pull Request Review – Enhancing Home Page with Skills & Country Cards

### ✅ What's Good

- **✅ Modular Components**: `ITJobsByCountryCard.jsx` and `SkillsChart.jsx` are nicely separated and scoped. This modularity makes the codebase cleaner and easier to maintain.
- **✅ Visual Enhancement**: Integrating data visualizations via Recharts significantly improves the user interface and experience.
- **✅ Tailwind Consistency**: Great to see consistent use of Tailwind CSS throughout – it keeps the UI unified.
- **✅ Clean Home Page Integration**: The new components are well-integrated into `HomePage.jsx`, with no clutter.

---

### Suggestions for Improvement

####  Error Handling & Loading States

Neither component currently handles loading, error, or empty data states.

**Suggested Improvement**:
```jsx
if (isLoading) return <div>Loading...</div>;
if (error) return <div>Error loading data.</div>;
if (!data?.length) return <div>No data available.</div>;

please get these done and let us know, cheers

SupaHotBall committed:
Fix: Implement loading, error, and empty state handling in ITJobsByCountryCard and SkillsChart components

SupaHotBall commented:
##  Changes Made

###  `ITJobsByCountryCard.jsx`
- Added placeholder flags for `isLoading` and `error`
- Render fallback messages for:
  - Loading state
  - Error state
  - No data available

###  `SkillsChart.jsx`
- Added checks for:
  - `isLoading`
  - `error`
  - Empty data array
- Message includes dynamic chart title for clearer UX feedback

AhmadShoaib1 reviewed (approved):
thank you adding the error handling function to your code, they work well., approved and merged.
